core/src/Kokkos_MemoryPool.hpp

```diff
@@ -165,8 +165,10 @@ class MemoryPool {
         accessible ? m_sb_state_array : (uint32_t *)host.allocate(alloc_size);
 
     if (!accessible) {
-      Kokkos::Impl::DeepCopy<Kokkos::HostSpace, base_memory_space>(
-          sb_state_array, m_sb_state_array, alloc_size);
+      Kokkos::Impl::DeepCopy<Kokkos::HostSpace, base_memory_space,
+                             typename base_memory_space::execution_space>(
+          typename base_memory_space::execution_space{}, sb_state_array,
```

Comment:
This changes the execution space template argument.
See https://github.com/kokkos/kokkos/blob/baba8ca88410ecccf9144af18f1fa0f18203ff48/core/src/Kokkos_Core_fwd.hpp#L282-L285

Is that intentional?

Reply:
This was just to make the code compile. I'll use `DstSpace::execution_space` then (i.e. `Kokkos::DefaultHostExecutionSpace`).